tests/facility/settings/activityDefinition/activityDefinitionEdit.spec.ts

```diff
@@ -0,0 +1,422 @@
+    await selectFromCategoryPicker(page, chargePicker, {
+      closeAfterSelect: true,
+      navigateCategories: [selectedCategory],
+      search: selectedChargeItem,
+    });
+
+    const locationsSection = page.getByText(/^locations$/i).locator("..");
+    await locationsSection.scrollIntoViewIfNeeded();
+    const locationsTrigger = locationsSection.getByRole("combobox").first();
+    await selectFromLocationMultiSelect(page, locationsTrigger, {
+      search: selectedLocation,
+    });
+
+    const diagSection = page
+      .getByText(/^diagnostic report codes$/i)
+      .locator("..");
+    const diagCombobox = diagSection.getByRole("combobox").first();
+    await selectFromValueSet(page, diagCombobox, {
+      search: selectedDiagCode,
+    });
+
+    await page.getByRole("button", { name: "Save" }).click();
+    await expectToast(page, /activity definition updated successfully/i);
+
+    // Now navigate back to edit and verify all fields are prefilled
+    await page.goto(`/facility/${facilityId}/settings/activity_definitions`);
+    await page.getByText(resourceCategoryName).click();
+
+    await clearFilter(page);
+
+    const searchInput2 = page.getByPlaceholder(/search/i);
+    await searchInput2.fill(createdAD.title);
+
+    const activityRow2 = page.locator("tr", { hasText: createdAD.title });
+    await expect(activityRow2).toBeVisible();
+    await activityRow2.getByRole("link", { name: /view/i }).click();
+
+    await page.getByRole("button", { name: /edit/i }).click();
+
+    await expect(page.getByLabel(/title.*\*/i)).toHaveValue(createdAD.title);
+    const expectedSlug = generateExpectedSlug(createdAD.title);
+    await expect(page.getByLabel(/slug/i)).toHaveValue(expectedSlug);
+    await expect(page.getByLabel(/description.*\*/i)).toHaveValue(
+      createdAD.description,
+    );
+    await expect(page.getByLabel(/usage.*\*/i)).toHaveValue(createdAD.usage);
+
+    await expect(page.getByLabel(/^status$/i)).toContainText(
+      new RegExp(createdAD.status, "i"),
+    );
+    await expect(
+      page.getByRole("combobox", { name: "Category" }),
+    ).toContainText(
+      new RegExp(createdAD.classification.replace(/_/g, "\\s"), "i"),
+    );
+    await expect(page.getByLabel(/^kind$/i)).toContainText(/service request/i);
+
+    const codeCombobox2 = page.getByRole("combobox", { name: /^code/i });
+    await expect(codeCombobox2).toContainText(selectedCode);
+
+    await expect(page.getByLabel(/^derived from uri$/i)).toHaveValue(
+      createdAD.derivedFromUri,
+    );
+
+    const bodySiteCombobox = page.getByRole("combobox", {
+      name: /body site/i,
+    });
+    await expect(bodySiteCombobox).toContainText(selectedBodySite);
+
+    const specimenContainer2 = page
+      .getByText(/^specimen requirements$/i)
+      .locator("..");
+    await expect(
+      specimenContainer2.getByText(selectedSpecimen).first(),
+    ).toBeVisible();
+
+    const obsContainer2 = page
```

Comment:
- why are you using a const for a single action, cant you directly use it and you proper location